test/integration/targets/include_vars/tasks/main.yml

```
@@ -223,6 +223,66 @@
       - "config.key2.b == 22"
       - "config.key3 == 3"
 
+- name: Include first vars file to test keep_existing
+  ansible.builtin.include_vars:
+    file: "{{ role_path }}/vars2/keep/existing.yml"
+    keep_existing: true
+
+- name: Verify that the vars were included
+  assert:
+    that:
+      - "keep_test_a == 1"
+      - "keep_test_b == false"
+      - "keep_test_c == 3"
+      - "keep_test_d == {'var1': 'foo', 'var2': 42}"
+
+- name: Include second vars file to test keep_existing
+  ansible.builtin.include_vars:
+    file: "{{ role_path }}/vars2/keep/new.yml"
+    keep_existing: true
+
+- name: Verify that the existing vars are kept
+  assert:
+    that:
+      - "keep_test_a == 1"
+      - "keep_test_b == false"
+      - "keep_test_c == 3"
+      - "keep_test_d == {'var1': 'foo', 'var2': 42}"
+      - "keep_test_e == '...'"
+
+- name: Include directory to test keep_existing
+  ansible.builtin.include_vars:
+    dir: "{{ role_path }}/vars2/keep2/"
+    keep_existing: true
+
+- name: Verify that only the vars from existing.yml are defined
+  assert:
+    that:
+      - "keep2_test_a == 1"
+      - "keep2_test_b == false"
+      - "keep2_test_c == 3"
+      - "keep2_test_d == {'var1': 'foo', 'var2': 42}"
+      - "keep2_test_e == '...'"
+
+- name: Include a vars file with a hash variable to test keep_existing
+  include_vars:
+    file: vars2/hashes/hash1.yml
+
+- name: Include the second file to merge the hash variable with keep_existing
+  include_vars:
+    file: vars2/hashes/hash2.yml
+    keep_existing: true
+    hash_behaviour: merge
+
+- name: Verify that the first hash variable is kept
+  assert:
+    that:
+      - "{{ config | length }} == 3"
+      - "config.key0 == 0"
+      - "config.key1 == 0"
+      - "{{ config.key2 | length }} == 1"
+      - "config.key2.a == 21"
```

Don't nest moustaches. `that` takes a list of Jinja expressions, so it should not contain additional Jinja delimiters.
```suggestion
      - config | length == 3
      - config.key0 == 0
      - config.key1 == 0
      - config.key2 | length == 1
      - config.key2.a == 21
```

---

Makes sense.
I corrected that, also for the other asserts along the way.